src/cmap/connect.ts

```diff
@@ -381,10 +415,81 @@ function makeConnection(options: ConnectionOptions, _callback: CallbackWithType<
     options.cancellationToken.once('cancel', cancellationHandler);
   }
 
-  socket.once(connectEvent, connectHandler);
+  if (existingSocket) {
+    process.nextTick(connectHandler);
+  } else {
+    socket.once(connectEvent, connectHandler);
+  }
+}
+
+function makeSocks5Connection(
+  options: MakeConnectionOptions & { proxyOptions: undefined },
+  proxyOptions: ProxyOptions,
+  callback: Callback<Stream>
+) {
+  const proxyHost = proxyOptions.host.includes(':') ? `[${proxyOptions.host}]` : proxyOptions.host;
+  const proxyPort = proxyOptions.port ?? 1080;
+  const hostAddress = HostAddress.fromString(`${proxyHost}:${proxyPort}`);
+
+  // First, connect to the proxy server itself:
+  makeConnection(
+    {
+      ...options,
+      hostAddress,
+      tls: false
+    },
+    (err, rawSocket) => {
+      if (err) {
+        return callback(err);
+      }
+
+      const destination = parseConnectOptions(options) as net.TcpNetConnectOpts;
+      if (typeof destination.host !== 'string' || typeof destination.port !== 'number') {
+        return callback(
+          new MongoInvalidArgumentError('Can only make Socks5 connections to TCP hosts')
+        );
+      }
+
+      // Then, establish the Socks5 proxy connection:
+      SocksClient.createConnection(
+        {
+          existing_socket: rawSocket,
+          timeout: options.connectTimeoutMS,
+          command: 'connect',
+          destination: {
+            host: destination.host,
+            port: destination.port
+          },
+          proxy: {
+            // host and port are ignored because we pass existing_socket
+            host: 'localhost',
```

Comment:
Weird that we have to defined this, but an opportunity to put any random string here to make it clear its ignored like `'iLoveJavaScript'`! 
We can totally leave this alone.. but why not ship an easter egg? :)